test/gtest/uct/test_uct_ep.cc

```diff
@@ -85,4 +162,73 @@ UCS_TEST_SKIP_COND_P(test_uct_ep, disconnect_after_send,
     }
 }
 
+UCS_TEST_SKIP_COND_P(test_uct_ep, destroy_entity_after_send,
+                     !check_caps(UCT_IFACE_FLAG_AM_ZCOPY))
+{
+    const unsigned max_iter = 300 / ucs::test_time_multiplier();
+    char *buffer_mem        = new char[sizeof(mapped_buffer)];
+
+    for (unsigned i = 0; i < max_iter; ++i) {
+        connect(false);
+
+        mapped_buffer *buffer = new (buffer_mem)
```

Comment:
why need to dynamic allocate the mapped_buffer?

Reply:
can we use ucs::scoped_ptr

Reply:
there is no `ucs::scoped_ptr` or am I missing smth?
done by allocating `mapped_buffer` in `{ }` scope

Reply:
sorry, i've meant ucs::auto_ptr